src/storage/txn/process.rs

```diff
@@ -134,31 +134,28 @@ pub(super) fn process_read_impl<E: Engine>(
             let result = reader.scan_locks(
                 scan_key.as_ref(),
                 |lock| txn_status.contains_key(&lock.ts),
-                RESOLVE_LOCK_BATCH_SIZE,
+                RESOLVE_LOCK_BATCH_SIZE + 1, // +1 is for scan out the next lock to be scanned
             );
             statistics.add(reader.get_statistics());
-            let (kv_pairs, has_remain) = result?;
+            let (mut kv_pairs, _) = result?;
+            let has_remain = kv_pairs.len() > RESOLVE_LOCK_BATCH_SIZE;
             tls_collect_keyread_histogram_vec(tag.get_str(), kv_pairs.len() as f64);
 
             if kv_pairs.is_empty() {
                 Ok(ProcessResult::Res)
             } else {
-                let next_scan_key = if has_remain {
-                    // There might be more locks.
-                    kv_pairs.last().map(|(k, _lock)| k.clone())
-                } else {
-                    // All locks are scanned
-                    None
-                };
-                Ok(ProcessResult::NextCommand {
-                    cmd: ResolveLock::new(
-                        mem::take(txn_status),
-                        next_scan_key,
-                        kv_pairs,
-                        ctx.clone(),
-                    )
-                    .into(),
-                })
+                let mut cmds = Vec::with_capacity(2);
+                if has_remain {
+                    let next_scan_key = kv_pairs.pop().map(|(k, _lock)| k);
+                    cmds.push(
+                        ResolveLockReadPhase::new(txn_status.clone(), next_scan_key, ctx.clone())
```

Comment:
Now `txn_status` can be shared in multiple commands. Is it a good idea to use an `Arc` to save some clones?